### PR TITLE
Dblatcher upgrade bridget

### PR DESCRIPTION
**Pull request by dblatcher**
## What does this change?
- upgrades the bridget dependency to the new [version 2.0.0](https://www.npmjs.com/package/@guardian/bridget/v/2.0.0)
- add the new Newsletters service to the native API and uses it in the `signUpForm.ts` module instead of the placeholder function

## Why?
The native functionality to sign up a user for a newsletter is being added (currently in development). This is PR hooks the new Bridget method into the script for the EmailSignupForm component (not currently being used - see https://github.com/guardian/dotcom-rendering/pull/5868).

## Follow-up work on later PR
 - Use the 'Environment' native API client side to determine the version of bridget in use
 - Conditionally render the `signUpForm.ts` - should only display if the app has Bridget 2.0.0 or higher running.

### CODE REVIEW

File: apps-rendering/src/client/signupForm.ts [hunk 1]
```diff
@@ -31,13 +31,22 @@ async function handleSubmission(bundle: FormBundle): Promise<void> {
...
+	} catch (error) {
+		success = false;
+	}
 	form.classList.remove(MODIFIER_CLASSNAME.waiting);
 
-	if (response.status === 200) {
+	if (success) {
 		form.classList.add(MODIFIER_CLASSNAME.success);
 	} else {
 		form.classList.add(MODIFIER_CLASSNAME.failure);
```
Comment (georgeblahblah) [inline apps-rendering/src/client/signupForm.ts:52]: Another way we could handle the potential throw (which we've used elsewhere in AR) is to use a `.then().finally` block:

```ts
	newslettersClient.requestSignUp(input.value, identityName).then(success => {
		if (success) {
			form.classList.add(MODIFIER_CLASSNAME.success);
		} else {
			form.classList.add(MODIFIER_CLASSNAME.failure);
		}
	})
	.finally(() => {
		form.classList.remove(MODIFIER_CLASSNAME.waiting);
	})
```

that way, we could also remove the local declaration of `success`!

**Review by georgeblahblah — COMMENTED** [review]
Looking good! I had one suggestion about using a promise `.then().finally` block to handle the throw a bit differently, but it's only a suggestion.

Otherwise, I think it might be good to rename the module from `signupForm` to `newsletterSignupForm` to make it clear what it relates to.

File: apps-rendering/src/client/signupForm.ts (hunk 1 above)
Comment (dblatcher) [reply apps-rendering/src/client/signupForm.ts]: thanks - that looks much better! One change I made was to include a `catch` statement to do `form.classList.add(MODIFIER_CLASSNAME.failure)` so the UI will be updated to the fail state if the function throws, as well as if it returns false.

**Comment by dblatcher** [comment]
> Otherwise, I think it might be good to rename the module from `signupForm` to `newsletterSignupForm` to make it clear what it relates to.
👍

**Review by georgeblahblah — APPROVED** [review]
Looking good! It might be worth testing this on CODE against a version of the app with an older version of Bridget. I'm happy to help support this if it's useful!

**Pull request merged**